### PR TITLE
Add check on nutrient values entered

### LINKED ISSUE #1857: Add maximum to nutrition values
Add maximum to nutrition values. Tacite reported values of 10000 kcal inputed by users.

### PR DESCRIPTION
**Pull request by KarthikRIyer**
## Description

A cap has been added to the nutrient values. When a value goes out of bounds a feedback will be displayed above the field.

## Related issues and discussion
closes #1857 
 
 ## Screen-shots, if any
whatsapp image 2019-01-17 at 1 40 01 pm

### CODE REVIEW

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 1]
```diff
@@ -424,15 +536,31 @@
...
+            <TextView
+                android:id="@+id/salt_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 2]
```diff
@@ -462,15 +590,31 @@
...
+            <TextView
+                android:id="@+id/sodium_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 3]
```diff
@@ -386,15 +482,31 @@
...
+            <TextView
+                android:id="@+id/proteins_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 4]
```diff
@@ -348,15 +428,31 @@
...
+            <TextView
+                android:id="@+id/dietary_fibre_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 5]
```diff
@@ -310,15 +374,31 @@
...
+            <TextView
+                android:id="@+id/sugar_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/layout/fragment_add_product_nutrition_facts.xml [hunk 6]
```diff
@@ -272,15 +320,31 @@
...
+            <TextView
+                android:id="@+id/carbohydrate_msg"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:layout_marginEnd="85dp"
+                android:layout_marginRight="85dp"
+                android:text="Value entered is more than given per serving size"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_add_product_nutrition_facts.xml]: Can you extract the string ?

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -871,6 +871,8 @@
     <string name="efsa_warning_high_risk">The European Food Safety Authority (EFSA) has determined that some population groups have a high risk of consuming too much %1$s.</string>
     <string name="legend_noael">above dose with observed adverse effect (NOAEL)</string>
     <string name="legend_adi">above Acceptable Daily Intake (ADI)</string>
+    <string name="max_energy_val_msg">Max value can be 2000kcal/100g</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="max_energy_val_msg">This value is above 2000 kcal per 100g. Can you correct it ?</string>
```

**Comment by PrajwalM2212** [comment]
@KarthikRIyer  Thanks

**Pull request merged**